core/CliMulti/Process.php

```diff
@@ -210,7 +210,9 @@ public static function isSupported()
             return false;
         }
 
-        if (!self::commandExists('ps') || !self::returnsSuccessCode('ps') || !self::commandExists('awk')) {
+        $awkTestCommand = 'awk -Wversion 2>/dev/null || awk --version || awk \'BEGIN {print "ping"}\' || awk';
```

Comment:
The `returnsSuccessCode()` actually appends a `> /dev/null 2>&1; echo $?` to the provided command. shouldn't the commands we provide here be wrapped in brackets so the appended `> /dev/null 2>&1` does not only apply to the last command?

Reply:
btw @diosmosis should be fine detecting AWK like this. Be good to check out comment from @sgiehl 

Had just another thought before why not put the command we use `awk '! /defunct/ {print $1}` in a constant and check if we get expected output?

like run `echo "  537 s000  Ss     0:00.05 login -pfl theuser /bin/bash -c exec -la bash /bin/bash" | awk '! /defunct/ {print $1}'` we expect to get back `537`? This way we would know if the command is there and behaves the way we expect maybe? Not 100% sure if it's a good idea but this way might be best to detect if archiving will work. Of course no longer such a good idea if we were to use multiple / different `awk` commands.

Same could be maybe done for `ps` instead of checking if command exists we use `ps x 2` and we check whether we get a 0 exit code since this is also what the class actually executes. I remember in past there were sometimes issues with the arguments where PS worked but not specific arguments and this way maybe would detect if we get what we expect or so.

Or maybe even easier could we simply execute `getListOfRunningProcesses` and/or `getRunningProcesses` and check whether we get a 0 response code (can't check for !empty maybe if no processes are detected). Just a thought to maybe simplify this.

Reply:
@tsteur @sgiehl updated the check + removed the `||` use.